### PR TITLE
New driver for ILI9341-OLED

### PR DESCRIPTION
**Pull request by Mum-Pf**
New driver for ILI9341-OLED-colordisplay, with rotation, flip, colordepth, gammacorrection. Works also in recovery.

Tested with 2.8inch OLED-Display ILI9241

/squeezelite-esp32/components/display/display.c (added ILI9341_Detect)
/squeezelite-esp32/components/display/ILI9341.c (new driver for ILI9341-Display... template was ST77xx.c)

To use it with recovery, compile recovery.bin and flash it too.

### CODE REVIEW

File: components/display/ILI9341.c [hunk 1]
```diff
@@ -10,356 +11,392 @@
...
+	}
+	if (VFlip) {						//Orientation 270 degree
+		Private->MADCtl = 0xE0;
+		int a = Device->Height;
+		int b = Device->Width;
+		Device->Height = b;
+		Device->Width = a;
+	}
+	*/
+
```
Comment (philippe44) [inline components/display/ILI9341.c:254]: Do we need to keep these comments? I'm always trying to remove dead code unless we really want to keep it

File: components/display/ILI9341.c [hunk 2]
```diff
@@ -10,356 +11,392 @@
...
 	
+	// Sleepout + Booster
+	Device->WriteCommand( Device, 0x11 );
+		
+	// need BGR & Address Mode
+	//Private->MADCtl = 1 << 3;		// for ST77xx = 0x40
+	//Private->MADCtl = 1 << 7;		// for ILI9341 = 0x80 (320x240) or 0x20 (240x320)
+	//Device->WriteCommand( Device, 0x36 );
+	//WriteByte( Device, Private->MADCtl );	
+
```
Comment (philippe44) [inline components/display/ILI9341.c:302]: Same

File: components/display/ILI9341.c [hunk 3]
```diff
@@ -10,356 +11,392 @@
...
+	 Device->WriteCommand( Device, 0xE0 );								// Positive Gamma Correction (15 Parameter)
+	 WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 );
+	 WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 );
+	 WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 ); WriteByte( Device, 0x22 );
+	Device->WriteCommand( Device, 0xE1 ); 								// Negative Gamma Correction (15 Parameter)
+	 WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 );
+	 WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 );
+	 WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 ); WriteByte( Device, 0x10 );
+	 */
+
```
Comment (philippe44) [inline components/display/ILI9341.c:360]: Same - there are also small indent misalignement

File: components/display/ILI9341.c [hunk 4]
```diff
@@ -10,356 +11,392 @@
...
-	.Update = Update, .Init = Init,
-};	
+	.Update = Update16, .Init = Init,
+	.Mode = GDS_RGB565, .Depth = 16,
+};		
 
 struct GDS_Device* ILI9341_Detect(char *Driver, struct GDS_Device* Device) {
-	if (!strcasestr(Driver, "ILI9341")) return NULL;
+	uint8_t Model;
+	//int Depth=18;		// 18bit (=24bit) colordepth
```
Comment (philippe44) [inline components/display/ILI9341.c:377]: same

**Comment by Mum-Pf** [comment]
You are welcome to delete all comments, they are not absolutely necessary. They are only help for me... thank you for review.

**Comment by philippe44** [comment]
Thanks!

**Pull request merged**